ui/pages/confirmations/hooks/alerts/transactions/useMultipleApprovalsAlerts.ts

```diff
@@ -0,0 +1,164 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { Hex } from '@metamask/utils';
+import { BigNumber } from 'bignumber.js';
+import { useMemo } from 'react';
+import { parseApprovalTransactionData } from '../../../../../../shared/modules/transaction.utils';
+import { RowAlertKey } from '../../../../../components/app/confirm/info/row/constants';
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { useConfirmContext } from '../../../context/confirm';
+
+type ApprovalInfo = {
+  tokenAddress: Hex;
+  spenderAddress?: Hex;
+  amount?: BigNumber;
+  isNFT: boolean;
+};
+
+type TokenOutflow = {
+  tokenAddress: Hex;
+  amount: BigNumber;
+  isDecrease: boolean;
+};
+
+export function useMultipleApprovalsAlerts(): Alert[] {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+
+  const isBatchTransaction = currentConfirmation?.type === 'batch';
+  const nestedTransactions = currentConfirmation?.nestedTransactions;
+  const simulationDataArray =
+    currentConfirmation?.simulationData?.tokenBalanceChanges;
+
+  // Extract all approvals from nested transactions
+  const approvals = useMemo((): ApprovalInfo[] => {
+    if (!isBatchTransaction || !nestedTransactions) {
+      return [];
+    }
+
+    const approvalsList: ApprovalInfo[] = [];
+
+    nestedTransactions.forEach((transaction) => {
```

Comment:
Can we simplify this by re-using `useBatchApproveBalanceChanges.ts` to just return us a list of parsed approvals?

Reply:
We now use useBatchApproveBalanceChanges.